src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -2067,4 +2068,16 @@ public static void OutputTrace(string s)
 
         public static string IdOf(object o) => o == null ? "(null)" : $"{o.GetType().Name}#{o.GetHashCode():X2}";
     }
+
+    // struct wrapper is used in order to improve the performance of the epoll thread hot path by up to 3% of some TechEmpower benchmarks
+    // the goal is to have a dedicated generic instantiation and using:
+    // System.Collections.Dictionary`2[System.IntPtr,System.Net.Sockets.SocketAsyncContextWrapper]::TryGetValueInternal(!0,int32,!1&)
+    // instead of:
+    // System.Collections.Dictionary`2[System.IntPtr,System.__Canon]::TryGetValueInternal(!0,int32,!1&)
+    internal readonly struct SocketAsyncContextWrapper
```

Comment:
nit: since this is only used in the SocketAsyncEngine as an optimization, maybe the struct can live in that file.